symbolic-debuginfo/src/sourcebundle.rs

```diff
@@ -660,6 +757,32 @@ impl<'data> SourceBundleDebugSession<'data> {
         let content = self.source_by_zip_path(zip_path)?;
         Ok(content.map(|opt| SourceCode::Content(Cow::Owned(opt))))
     }
+
+    /// Looks up some source by debug ID and file type.
+    ///
+    /// Lookups by [`DebugId`] require knowledge of the file that is supposed to be
+    /// looked up as multiple files (one per type) can share the same debug ID.
+    /// Special care needs to be taken about [`SourceFileType::IndexedRamBundle`]
+    /// and [`SourceFileType::SourceMap`] which are different file types despite
+    /// the name of it.
+    ///
+    /// # Note on Abstractions
+    ///
+    /// This method is currently not exposed via a standardized debug session
+    /// as it's primarily used for the JavaScript processing system which uses
+    /// different abstractions.
+    pub fn source_by_debug_id(
```

Comment:
It's a bit odd that we don't actually ever expose the `SourceFileInfo` via the debug session. This is not necessarily a problem but it means that at least on the rust consumer site, you never manage to get access to the headers and with that the source mapping URL. I suppose this will need to be changed once symbolicator actually starts processing this.

cc @kamilogorek @Swatinem

Reply:
I would propose we change this entire interface to return an `Option<SourceDescriptor<'_>>` instead which contains `url`, `contents`, `debug_id` and whatever. This refs #758

Reply:
By "this entire interface", do you also mean the `source_by_path` method on `DebugSession`?

Reply:
Correct. I would change the return value in either case to be a full descriptor of the source, not just the containing bytes.

Reply:
I've opened a WIP PR for this: https://github.com/getsentry/symbolic/pull/763

Reply:
@loewenheim ugh this badly overlapped with me addressing this in this PR now.